octree/include/pcl/octree/octree_iterator.h

```diff
@@ -568,6 +568,95 @@ namespace pcl
         std::deque<IteratorState> FIFO_;
       };
 
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    /** \brief @b Octree iterator class
+     * \note Iterator  over all existing nodes at a given depth. It walks across an octree
+     *       in a breadth-first manner.
+     * \ingroup octree
+     * \author Fabien Rozar (fabien.rozar@gmail.com)
+     */
+    template<typename OctreeT>
+    class OctreeFixedDepthIterator : public OctreeBreadthFirstIterator<OctreeT>
+    {
+    public:
+
+      // public typedefs
+      using typename OctreeBreadthFirstIterator<OctreeT>::BranchNode;
+      using typename OctreeBreadthFirstIterator<OctreeT>::LeafNode;
+
+      /** \brief Empty constructor.
```

Comment:
There is no parameter.  Also, "emtpy constructor" is not very helpful. We should rather say that the constructed iterator is not valid. Finally, `explicit` is not needed.

Reply:
The empty ctor call `OctreeBreadthFirstIterator<OctreeT> (0u)` which put the iterator in end state. 

The `this->reset (0u);` has been move from this empty ctor (its body is really empty now).